pkg/policyconditions/conditions.go

```diff
@@ -119,7 +119,7 @@ func Update(cli client.Client, apiReader client.Reader, policyKey types.Namespac
 	return err
 }
 
-func update(apiWriter client.Client, apiReader client.Reader, policyReader client.Reader, policyKey types.NamespacedName) error {
+func update(apiWriter client.Client, apiReader, policyReader client.Reader, policyKey types.NamespacedName) error {
```

Comment:
I agree. Personally, I'd consider removing this check altogether. I think that this does not improve readability that much, maybe it is even a bit more complex this way, because you have to go back and look for variable type.